main/src/ca/uwaterloo/flix/language/phase/jvm/AsmOps.scala

```diff
@@ -120,19 +122,50 @@ object AsmOps {
   }
 
   /**
-    * Returns the array type code for the value of the type specified by `tpe`
+    * Returns the array store instruction for arrays of the given JvmType tpe
     */
-  def getArrayTypeCode(tpe: JvmType): Int = tpe match {
-    case JvmType.Void => throw InternalCompilerException(s"Unexpected type $tpe", SourceLocation.Unknown)
-    case JvmType.PrimBool => T_BOOLEAN
-    case JvmType.PrimChar => T_CHAR
-    case JvmType.PrimFloat => T_FLOAT
-    case JvmType.PrimDouble => T_DOUBLE
-    case JvmType.PrimByte => T_BYTE
-    case JvmType.PrimShort => T_SHORT
-    case JvmType.PrimInt => T_INT
-    case JvmType.PrimLong => T_LONG
-    case JvmType.Reference(_) => throw InternalCompilerException(s"Expected primitive type. Actual type: $tpe", SourceLocation.Unknown)
```

Comment:
Hmm maybe we should add an intermediate interface `Primitive` between `BackendType` and all the primitives.

Reply:
Do you think we should add that in `BackendType` or what role do you imagine `Primitive` having?

Reply:
I assume it should be a `case object` that extends `BackendType` if we just want it to wrap all the usages as is.

Reply:
The primitive type could actually implement most of the functions in `AsmOps`.

Reply:
yeah, but leave that to another PR